anyway/flask_app.py

```diff
@@ -1502,95 +1499,77 @@ def logout():
 app.add_url_rule("/api/news-flash", endpoint=None, view_func=news_flash, methods=["GET"])
 
 
-def check_is_a_safe_redirect_url(url: str) -> bool:
-    url_obj = urlparse(url)
-    if url_obj.scheme not in ['https', 'http']:
-        return False
-
-    netloc = url_obj.netloc
-    if not netloc:
-        return False
-
-    # Note that we don't support ipv6 localhost address or ipv4 localhost full range of address
-    if netloc in ["www.anyway.co.il", "anyway-infographics-staging.web.app", "localhost", "127.0.0.1"]:
-        return True
-    else:  # Check localhost with port
-        localhost_regex = re.compile(r'^127\.0\.0\.1\:[0-9]{1,7}$|^localhost\:[0-9]{1,7}$')
-        if localhost_regex.match(netloc):
-            return True
-
-    return False
-
-
 @app.route("/authorize/<provider>")
-def oauth_authorize(provider):
-    # This is quick fix for the DEMO - to make sure that anyway.co.il is fully compatible with https://anyway-infographics-staging.web.app/
-    redirect_url_from_url = request.args.get('redirect_url', type=str)
+def oauth_authorize(provider: str) -> Response:
+    if provider != "google":
+        return Response(
+            "Google is the only support OAuth 2.0 provider.",
+            status=422,
+        )
+    if not current_user.is_anonymous:
+        return Response(
+            "User is already login.",
+            status=400,
+        )
+
+    redirect_url_from_url = request.args.get("redirect_url", type=str)
     redirect_url = BE_CONST.DEFAULT_REDIRECT_URL
     if redirect_url_from_url and check_is_a_safe_redirect_url(redirect_url_from_url):
         redirect_url = redirect_url_from_url
 
-    if provider != "google":
-        return redirect(redirect_url, code=302)
-    if not current_user.is_anonymous:
-        return redirect(redirect_url, code=302)
-
     oauth = OAuthSignIn.get_provider(provider)
     return oauth.authorize(redirect_url=redirect_url)
 
 
 @app.route("/callback/<provider>")
-def oauth_callback(provider):
-    # This is quick fix for the DEMO - to make sure that anyway.co.il is fully compatible with https://anyway-infographics-staging.web.app/
+def oauth_callback(provider: str) -> Response:
+    if provider != "google":
+        return Response(
+            "Google is the only support OAuth 2.0 provider.",
```

Comment:
```suggestion
            "Google is the only supported OAuth 2.0 provider.",
```

Or maybe just `"Unsupported OAuth 2.0 provider."